templates/repo/sub_menu.tmpl

```diff
@@ -15,7 +15,9 @@
 					</div>
 				{{end}}
 				<div class="item">
-					<span>{{svg "octicon-database"}} <b>{{FileSize .Repository.Size}}</b></span>
+					{{$fileSizeFormatted := FileSize .Repository.Size}}{{/* the formatted string is always "{val} {unit}" */}}
+					{{$fileSizeFields := StringUtils.Split $fileSizeFormatted " "}}
+					<span>{{svg "octicon-database"}} <b>{{index $fileSizeFields 0}}</b> {{index $fileSizeFields 1}}</span>
```

Comment:
It doesn't work because it's a string , not a number.

![image](https://user-images.githubusercontent.com/2114189/235207587-e0460637-d97c-456a-bdb8-e8465ddb7281.png)

Reply:
Gotcha, nevermind then

Reply:
And luckily, this number won't be too large, no "1024 B", it just becomes "1 KiB"

Reply:
How about:
```diff
diff --git a/modules/templates/util_string.go b/modules/templates/util_string.go
index 42d11fc99..1481f90f2 100644
--- a/modules/templates/util_string.go
+++ b/modules/templates/util_string.go
@@ -3,7 +3,10 @@
 
 package templates
 
-import "strings"
+import (
+	"strconv"
+	"strings"
+)
 
 type StringUtils struct{}
 
@@ -22,3 +25,8 @@ func (su *StringUtils) Contains(s, substr string) bool {
 func (su *StringUtils) Split(s, sep string) []string {
 	return strings.Split(s, sep)
 }
+
+func (su *StringUtils) ConvertToInt(s string) int {
+	num, _ := strconv.Atoi(s)
+	return num
+}
diff --git a/templates/repo/sub_menu.tmpl b/templates/repo/sub_menu.tmpl
index 3ef8adbe4..0f4d9384a 100644
--- a/templates/repo/sub_menu.tmpl
+++ b/templates/repo/sub_menu.tmpl
@@ -17,7 +17,7 @@
 				<div class="item">
 					{{$fileSizeFormatted := FileSize .Repository.Size}}{{/* the formatted string is always "{val} {unit}" */}}
 					{{$fileSizeFields := StringUtils.Split $fileSizeFormatted " "}}
-					<span>{{svg "octicon-database"}} <b>{{index $fileSizeFields 0}}</b> {{index $fileSizeFields 1}}</span>
+					<span>{{svg "octicon-database"}} <b>{{.locale.PrettyNumber (index $fileSizeFields 0 | StringUtils.ConvertToInt)}}</b> {{index $fileSizeFields 1}}</span>
 				</div>
 			{{end}}
 		</div>
```

Reply:
The `val` could also be float in some cases (see human.IBytes)

IMO not that necessary to make it use PrettyNumber, until some users requires that.

Reply:
If you really prefer, we can do this 2fa4cc3e7e35ee33f97e9ac1d8350c31bef9d898

ps: `int` can't be used, because a repo could be larger than 2G, then `int` overflows.